### ISSUE TITLE
Add button of repeatable components is missing a focus outline and focus handling

### ISSUE THREAD

**Issue by gu-stav** [strapi/strapi#15109]
### Required System information

- Node.js version: not applicable
- NPM version: not applicable
- Strapi version: 4.5.3
- Database: not applicable
- Operating system: MacOS

### Describe the bug

The "Add item" button of repeatable components has two a11y issues:

- focus outline is missing
- after item creation, the focus is not set on the item being created

https://user-images.githubusercontent.com/2244375/206165176-c29b7218-b51f-43aa-9f5f-6677d8fa9bfc.mp4

### Steps to reproduce the behavior

1. Create a content-type with a repeatable component
2. Tab to "Create item"
3. Hit enter

### Expected behavior

I expect a focus indicator to be visible. I expect the focus to be handled properly (on the created item?).

**Comment by BhuvaneshPatil** [comment]
I have investigated this issue. The problems I have found are the following and how they can be tackled.
- The *Add an Entry* buttons is being focussed, but the styling ( outlines/borders ) is not visible. The reason behind this behavior is, the child is completely fitting the parent ( which is flex ).

Here is the screenshot of the code - 

![Flex and Button Code](https://user-images.githubusercontent.com/27822551/206780163-ea5905f3-0983-4a81-8144-f773310cb9ba.png)


The good fix for this is to apply some padding to the parent.

![image](https://user-images.githubusercontent.com/27822551/206780378-7fba0de8-a1df-4a23-bdee-30fc3fdc920e.png)

For focussing on the input when we add input is I am exploring right now.

**Comment by gu-stav** [comment]
Fixed by the attached PR.
